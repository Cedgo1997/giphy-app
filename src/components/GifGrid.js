import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    useEffect(() => {
        getGifs();
    }, []);

    const [images, setImages] = useState([]);

    const getGifs = async () => {
        const apiKey = 'EfiDDliYy1sKOtC0ePfLVayOWsFVNM9x';
        const url = `https://api.giphy.com/v1/gifs/search?q=rick&limit=10&api_key=${apiKey}`;
        const resp = await fetch(url);
        const { data } = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });
        setImages(gifs);
    }

    return (
        <div>
            <h3>{category}</h3>
            <ol>
                {images.map((img) => 
                        <GifGridItem
                            img = {img}
                            {...img}
                        />)}
            </ol>
        </div>
    )
}
