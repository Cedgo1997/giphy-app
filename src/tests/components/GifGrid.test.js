import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";



jest.mock('../../hooks/useFetchGifs');
describe('Pruebas en GifGrid', () => {
    const category = 'Spider-Man';

    test('Debe renderizar correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
       const images = [{
           id: '1',
           url: 'https://localhost/cualquier/cosa.jpg',
           title: 'Cualquier cosa'
       }]
        useFetchGifs.mockReturnValue({
            data: images,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBeFalsy();
        expect(wrapper.find('GifGridItem').length).toBe(images.length);
    });
});