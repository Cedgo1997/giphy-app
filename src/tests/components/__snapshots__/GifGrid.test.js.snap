// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="card-grid animate__animated animate__fadeIn animate__delay-1s"
  >
    Spider-Man
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn animate__delay-1s"
  >
    <ul>
      <GifGridItem
        id="1"
        key="1"
        title="Cualquier cosa"
        url="https://localhost/cualquier/cosa.jpg"
      />
    </ul>
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid Debe renderizar correctamente 1`] = `
<Fragment>
  <h3
    className="card-grid animate__animated animate__fadeIn animate__delay-1s"
  >
    Spider-Man
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn animate__delay-1s"
  >
    <ul />
  </div>
</Fragment>
`;
